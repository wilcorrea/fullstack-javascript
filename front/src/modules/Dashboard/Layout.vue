<template>
  <div class="app-layout DashboardLayout">
    <!-- a fancy menu to app -->
    <app-menu v-bind="{ links }" />
    <!-- the main layout container -->
    <div class="app-layout__container">
      <!-- breadcrumb handle with $route directly -->
      <app-breadcrumb />
      <!-- where the magic happens -->
      <router-view />
    </div>
  </div>
</template>

<script lang="js">
/**
 */
export default {
  /**
   */
  name: 'DashboardLayout',
  /**
   */
  data: () => ({
    links: [
      { path: '/dashboard', label: 'In√≠cio' },
      { path: '/dashboard/orders', label: 'Pedidos' },
      { path: '/dashboard/about', label: 'Sobre' }
    ]
  }),
  /**
   */
  mounted () {
    this.$root.mainLayout = window.document.querySelector('.app-layout.DashboardLayout')
  }
}
</script>
