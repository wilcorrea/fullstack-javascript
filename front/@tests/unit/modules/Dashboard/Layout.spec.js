import { shallowMount } from '@vue/test-utils'
import Layout from 'src/modules/Dashboard/Layout.vue'

describe('Layout', () => {
  const AppBreadcrumb = { template: '<div>AppBreadcrumb</div>' }
  const AppMenu = { template: '<div>AppMenu</div>' }
  const RouterView = { template: '<div>RouterView</div>' }

  const wrapper = shallowMount(Layout, { components: { AppBreadcrumb, AppMenu, RouterView } })

  /**
   * @test
   */
  it('check component markup', () => {
    expect(wrapper.html()).toMatch('')
  })

  /**
   * @test
   */
  test('should render content correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
