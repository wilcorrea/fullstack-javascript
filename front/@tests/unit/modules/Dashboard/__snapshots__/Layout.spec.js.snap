// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render content correctly 1`] = `
<div class="app-layout DashboardLayout">
  <app-menu-stub links="[object Object],[object Object],[object Object]"></app-menu-stub>
  <div class="app-layout__container">
    <app-breadcrumb-stub></app-breadcrumb-stub>
    <router-view-stub></router-view-stub>
  </div>
</div>
`;
